### ISSUE TITLE
hints --type=linenum --linenum-action does no path path to editor

### ISSUE THREAD

**Issue by Domeee** [kovidgoyal/kitty#4675]
**Describe the bug**
Opening a file in neovim via hints mode opens the base path to the file correctly but passes only the +{line} option and ignores the path.

kitty.conf
```
map ctrl+shift+o kitten hints --type=linenum --linenum-action=window nvr --remote +{line} {path}
```
text acted on [1]
```
2pp/modules/footprintCalculator/components/stories/FootprintCalculator.stories.tsx:33:7 - error TS
2322: Type 'undefined' is not assignable to type 'number | null'.

33       selected_index: undefined,
         ~~~~~~~~~~~~~~

  1pp/modules/footprintCalculator/FootprintCalculator.ts:19:3
    19   selected_index: number | null;
         ~~~~~~~~~~~~~~
    The expected type comes from property 'selected_index' which is declared here on type 'Footpri
ntCalculatorField'
```
text acted on [2]
```
1home/dome/Code/enerjoy/code/enerjoy-mobile/app/modules/footprintCalculator/components/SelectListValueActionSheet.tsx:114:19: This assertion is unnecessary since it does not change the type of the exp
ression. [Error/@typescript-eslint/no-unnecessary-type-assertion]
```

path opened in neovim
```
~/Code/enerjoy/enerjoy-mobile/19
```
```
~/Code/enerjoy/enerjoy-mobile/114
```

Manually invoking the linenum-action works fine.

**To Reproduce**
1. act on the text provided with the provided kitty.conf

**Environment details**
```
kitty 0.24.2 created by Kovid Goyal
Linux encore 5.16.5-arch1-1 #1 SMP PREEMPT Tue, 01 Feb 2022 21:42:50 +0000 x86_64
Arch Linux 5.16.5-arch1-1 (/dev/tty)

LSB_VERSION=1.4
DISTRIB_ID=Arch
DISTRIB_RELEASE=rolling
DISTRIB_DESCRIPTION="Arch Linux"
Running under: X11
Frozen: False
Paths:
  kitty: /usr/bin/kitty
  base dir: /usr/lib/kitty
  extensions dir: /usr/lib/kitty/kitty
  system shell: /usr/bin/zsh
Loaded config files:
  /home/dome/.config/kitty/kitty.conf

Config options different from defaults:
clipboard_control    ('write-clipboard', 'write-primary')
cursor_shape         3
enable_audio_bell    False
font_size            12.0
scrollback_pager     ['nvim', '-c', 'set nonumber nolist showtabline=0 foldcolumn=0 laststatus=0', '-c', 'autocmd TermOpen * normal G', '-c', 'silent write! /tmp/kitty_scrollback_buffer | te head -c-1 /tmp/kitty_scrollback_buffer; rm /tmp/kitty_scrollback_buffer; cat']
window_padding_width FloatEdges(left=5.0, top=5.0, right=5.0, bottom=5.0)
Changed shortcuts:
	ctrl+shift+g → scroll_end
	ctrl+shift+o → kitten hints --type=linenum --linenum-action=window nvr --remote +{line} {path}
Colors:
	background           #282828   
	color0               #282828   
	color1               #f43753   
	color10              #c9d05c   
	color11              #ffc24b   
	color12              #b3deef   
	color13              #d3b987   
	color14              #73cef4   
	color15              #feffff   
	color2               #c9d05c   
	color3               #ffc24b   
	color4               #b3deef   
	color5               #d3b987   
	color6               #73cef4   
	color7               #eeeeee   
	color8               #4c4c4c   
	color9               #f43753   
	cursor               #eeeeee   
	foreground           #eeeeee   

Environment variable names seen by the kitty process:
	DBUS_SESSION_BUS_ADDRESS
	DISPLAY
	DOTNET_BUNDLE_EXTRACT_BASE_DIR
	DOTNET_ROOT
	GTK_MODULES
	HOME
	I3SOCK
	INVOCATION_ID
	LANG
	LOGNAME
	MAIL
	MOTD_SHOWN
	OLDPWD
	PATH
	PWD
	SHELL
	SHLVL
	SSH_AGENT_PID
	SSH_AUTH_SOCK
	SYSTEMD_EXEC_PID
	TERM
	USER
	WINDOWPATH
	XAUTHORITY
	XDG_DATA_DIRS
	XDG_RUNTIME_DIR
	XDG_SEAT
	XDG_SESSION_CLASS
	XDG_SESSION_ID
	XDG_SESSION_TYPE
	XDG_VTNR

```

**Comment by kovidgoyal** [comment]
That will be because whatever nvr is, is discarding the second argument.
Try running nvim or vim instead and it will work fine.

**Comment by Domeee** [comment]
Hi @kovidgoyal 

Thank you very much for your fast replay. As stated in the bug report, manually invoking the linenum-action works fine. It seems that kitty is not providing the {path} to the linenum-action.

**Comment by Domeee** [comment]
@kovidgoyal I think i found a possible problem. Typescript tsc and eslint both provide error messages as follows:

```
{path}:{line}:{column}
```

This does not seem to work with hints. The following text-format works:

```
{path}:{line}
```

Repro:

1. Run kitty as
kitty -o 'map ctrl+shift+o kitten hints --type=linenum --linenum-action=window vim --remote +{line} {path}'
2. Type setup.py:33:**1**
3. Press ctrl+shift+o
4. press 1
5. 33 is opened instead the full path including line

**Comment by kovidgoyal** [comment]
Does not repro for me with

```
kitty -o 'map ctrl+shift+o kitten hints --type=linenum --linenum-action=window vim --remote +{line} {path}' sh -c "echo setup.py:33:**1**; read"
```

or with

```
kitty -o 'map ctrl+shift+o kitten hints --type=linenum --linenum-action=window vim --remote +{line} {path}' sh -c "echo setup.py:33:1; read"
```

**Comment by Frydac** [comment]
Hi,

I bumped into this while after updating my system. {path} seems to be incorrect only if the string matched has a slash and line _and_ column in the following format:

`path/myfile.cpp:10:5`

I used this map to test with
```
map ctrl+shift+o kitten hints --type=linenum --linenum-action=self --hints-text-color=yellow path=\"{path}\" line=\"{line}\"
```
this prints with the above path as input (it highlights the path and both numbers)
`'path="10"' 'line="5"'`

while `path/myfile.cpp:10` and `myfile.cpp:10:5` work as expected 

In a previous version it used to also work as expected with `path/myfile.cpp:10:5` but I don't know what version that was.

(FYI: it would be nice to also have a {column} :) e.g. gcc/clang also provide that info in the above format)
